### ISSUE TITLE
Stream Data structure

### ISSUE THREAD

**Issue by aibarra11** [prwnr/laravel-streamer#15]
I'm trying to utilize the php artisan streamer:listen command. I have it configured in my config and have a class the implements Message receiver.

but, i keep getting on error

`Error: Undefined index: data`

traced it back to Streamer.php and its expecting a "data" index. My streams don't have that.

stream message example

`"XADD" "WorldRegionCreated" "*" "name" "Dominican Republic" "uuid" "f5650fee-8203-4774-b24e-2696d7cb9cc3"`

this was generated from:

````
namespace App\Messages;
use Prwnr\Streamer\Contracts\StreamableMessage;

class IncMessage implements StreamableMessage{
    public $data;

    public function setProps($model){
   
   $this->data = $model->getAttributes();
   unset($this->data['updated_at']);
   unset($this->data['created_at']);
   unset($this->data['id']);
    }

    
    public function getContent(): array
    {
        return $this->data;
    }

}
````

if i bypass your check i can easily die & dump the $payload and everything looks good.

my question - am i structuring my messages wrong? Or is it possible to remove the "data" index? 
thanks!

**Comment by prwnr** [comment]
@aibarra11 
my stream listening command basically expect a defined structure of stream event being emitted, as you faced. 
its not the `StreamableMessage` but its `Event` contract: https://github.com/prwnr/laravel-streamer/blob/master/examples/5_streaming.php

So, the emitting part (with use of Streamer facade) always expects this Event interface to be emitted and `Streamer` facade will shape it as it needs and Listener will rebuild it from stream without problems.

In other words: to be able to receive a message by my listener, you need to emit the message by my streamer.

**Comment by prwnr** [comment]
https://github.com/prwnr/laravel-streamer/blob/6001a73f0a6a954670f308dbb57be54e43e56693/src/EventDispatcher/Streamer.php#L119
see also this. here is where I shape event with meta data and the payload and after that in putting it on stream

**Comment by aibarra11** [comment]
thanks! one last question. The structure of your messages, is that pretty standard? We're developing an architecture for microservices for our company. Redis will be used as the messaging backbone. We're still in the early stages. Thanks!

````
dd($receivedMessage);

Prwnr\Streamer\EventDispatcher\ReceivedMessage^ {#794
  -id: "1573757562872-0"
  -content: array:8 [
    "_id" => "1573757562872-0"
    "type" => "event"
    "version" => "1.3"
    "name" => "PropertiesCreated"
    "domain" => "Laravel"
    "created" => "1573757349"
    "data" => array:6 [
      "name" => "Chile"
      "worldRegionUuid" => "b665393f-2336-4214-a064-28db2d567ef1"
      "uuid" => "d0759a0e-5e6d-44c7-8084-86db363e909c"
      "updated_at" => "2019-11-14 18:49:09"
      "created_at" => "2019-11-14 18:49:09"
      "id" => 61
    ]
    "hash" => "6511709c0d4b2eb0c962077e6cc8acc7d1ebadb0b483b10f1b36ae2c6556673f"
  ]
}

````

**Comment by prwnr** [comment]
@aibarra11 
yeah, what you posted is a standard event message structure and it will never be different (unless some breaking changes will come, but nothing planned yet).
`data` is your custom payload here, while the rest of the fields are populated on `Streamer::emit` call.
`_id` and `created` are what they say (id and emit timestamp), `type` depends on what you will specify in your Event method implementation, `name` is always the Event name - which is also stream name. `domain` can be changed in config, so that you can see for example from what service for comes. `hash` is a hash of all the other fields, for potential check if data matches something that you expect.

**Comment by aibarra11** [comment]
awesome, thanks!
